variable/include/scipp/variable/transform.h

```diff
@@ -459,23 +463,16 @@ template <bool dry_run> struct in_place {
 
     auto run = [&](auto &indices, const auto &end) {
       const auto inner_strides = indices.inner_strides();
-      if (std::all_of(inner_strides.begin(), inner_strides.end(),
-                      [](auto x) { return x == 0; })) {
-        // The special cases don't work when all strides are 0.
```

Comment:
Why is this not required any more?

Reply:
I honestly don't know. At first I thought it's not needed for non-in-place because `stride[0]==1` so I omitted the check. But the stride _can_ be zero and everything seems to work. So I checked the in-place case and all the tests pass without this check.

Reply:
Since we are unsure, I think we should actually just remove `{0}`, `{0,0}`, ... from the special cases. Cases where all strides are zero are just scalar and do not need the optimization anyway.

Reply:
I removed the cases to reduce the binary size / compile time.
But the check wasn't so much about the special cases but about how both inner loops works, including the fallback 
```.cpp
static void inner_loop(Op &&op, std::array<scipp::index, sizeof...(Operands)> indices,
           const std::array<scipp::index, sizeof...(Operands)> &strides,
           const scipp::index n, Operands &&... operands)
```
The reason why this used to fail is that `inner_size` (`n` in the function signature) used to be 0 for scalar operations. That is no longer the case since I changed to the more general 
```.cpp
const auto inner_size = indices.in_same_chunk(end, 1)
                                  ? indices.inner_distance_to(end)
                                  : indices.inner_distance_to_end();
```
which gives 1 for scalars. So the check is no longer required.